pyflow/RCDT2/Nodes/core.py

```diff
@@ -0,0 +1,127 @@
+from typing import Literal
+from PyFlow.Core import NodeBase, PinBase
+import rclpy
+from rclpy.node import Node
+from threading import Thread
+from logging import getLogger
+from inflection import underscore
+
+from sensor_msgs.msg import Image
+
+logger = getLogger(__name__)
+
+
+class PinManager:
+    def __init__(self):
+        self.input_names: list[str] = []
+        self.input_pins: list[PinBase] = []
+
+        self.output_names: list[str] = []
+        self.output_pins: list[PinBase] = []
+
+    def add_input_pin(self, name: str, pin: PinBase) -> None:
+        self.input_names.append(name)
+        self.input_pins.append(pin)
+
+    def add_ouput_pin(self, name: str, pin: PinBase) -> None:
+        self.output_names.append(name)
+        self.output_pins.append(pin)
+
+    def get_pin_names(self, pin_type: Literal["input", "output"]) -> list[str]:
+        return getattr(self, f"{pin_type}_names")
+
+
+class PyflowRosBridge:
+    rclpy.init()
+    node = Node("pyflow")
+    Thread(target=rclpy.spin, args=[node], daemon=True).start()
+    node.get_logger().info("Node started!")
+
+
+class PyflowBase(NodeBase):
+    def __init__(self, name: str):
+        super().__init__(name)
+
+        self.pin_manager = PinManager()
+
+    def get_data(self, pin_name: str) -> object:
+        self.pin_manager.get_pin(pin_name)
+
+
+class PyflowComputer(PyflowBase):
+    def __init__(self, name: str):
+        super().__init__(name)
```

Comment:
Een __init__() die alleen de super() aanroept is niet nodig, dit gebeurt altijd al als een class inherit van een parent class. Deze "lege" __init__() kan dus prima weggelaten worden